Tests/Opc.Ua.Client.Tests/ContinuationPointInBatchTest.cs

```diff
@@ -0,0 +1,1398 @@
+        [Theory]
+        [Test, Order(200)]
+        public void MBNodeCache_BrowseAllVariables_MultipleNodes(ManagedBrowseTestDataProvider testData)
+        {
+            Session theSession = ((Session)(((TraceableSession)Session).Session));
+            theSession.NodeCache.Clear();
+
+            // the ExpectedNumber* parameters are not relevant/correct for this test.
+            ManagedBrowseExpectedResultValues pass1ExpectedResults = new ManagedBrowseExpectedResultValues {
+                InputMaxNumberOfContinuationPoints = testData.MaxNumberOfContinuationPoints,
+                InputMaxNumberOfReferencesPerNode = testData.MaxNumberOfReferencesPerNode,
+                ExpectedNumberOfPasses = testData.ExpectedNumberOfPasses,
+                ExpectedNumberOfBadNoCPSCs = testData.ExpectedNumberOfBadNoCPSCs
+            };
+
+            ReferenceServerForThisUnitTest.Test_MaxBrowseReferencesPerNode =
+                pass1ExpectedResults.InputMaxNumberOfReferencesPerNode;
+
+            ReferenceServerForThisUnitTest.SetMaxNumberOfContinuationPoints(
+                pass1ExpectedResults.InputMaxNumberOfContinuationPoints);
+            theSession.ServerMaxContinuationPointsPerBrowse
+                = pass1ExpectedResults.InputMaxNumberOfContinuationPoints;
+
+            var result = new List<INode>();
+            var nodesToBrowse = new ExpandedNodeIdCollection {
+                ObjectIds.ObjectsFolder
+            };
+
+            Session.FetchTypeTree(ReferenceTypeIds.References);
+            var referenceTypeIds = new NodeIdCollection() { ReferenceTypeIds.HierarchicalReferences };
+            while (nodesToBrowse.Count > 0)
+            {
+                var nextNodesToBrowse = new ExpandedNodeIdCollection();
+                try
+                {
+                    var organizers = Session.NodeCache.FindReferences(
+                        nodesToBrowse,
+                        referenceTypeIds,
+                        false,
+                        true);
+                    nextNodesToBrowse.AddRange(organizers.Select(n => n.NodeId));
+                    var objectNodes = organizers.Where(n => n is ObjectNode);
+                    var variableNodes = organizers.Where(n => n is VariableNode);
+                    result.AddRange(variableNodes);
+                }
+                catch (ServiceResultException sre)
+                {
+                    if (sre.StatusCode == StatusCodes.BadUserAccessDenied)
+                    {
+                        TestContext.Out.WriteLine($"Access denied: Skipped node.");
+                    }
+                }
+                nodesToBrowse = new ExpandedNodeIdCollection(nextNodesToBrowse.Distinct());
+                TestContext.Out.WriteLine("Found {0} duplicates", nextNodesToBrowse.Count - nodesToBrowse.Count);
+            }
+
+            TestContext.Out.WriteLine("Found {0} variables", result.Count);
+        }
+
+
+        [DatapointSource]
```

Comment:
makes it easier to find when the datapointsource is defined in the top of the class, before setup.

Reply:
changed as requested.